src/mlpack/methods/ann/layer/softshrink.hpp

```diff
@@ -0,0 +1,208 @@
+                DataType&& gy,
+                DataType&& g);
+
+  //! Get the output parameter.
+  OutputDataType const& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the delta.
+  OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+  OutputDataType& Delta() { return delta; }
+
+  //! Get the hyperparameter lambda.
+  double const& Lambda() const { return lambda; }
+  //! Modify the hyperparameter lambda.
+  double& Lambda() { return lambda; }
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const unsigned int /* version */);
+
+ private:
+  /**
+   * Computes the value of Soft Shrink activation function.
+   *
+   * @param x Input data.
+   * @return f(x) 
+   */
+  double Fn(const double x)
+  {
+    if (x > lambda)
+    {
+      return x - lambda;
+    }
+    else if (x < -1 * lambda)
+    {
+      return x + lambda;
+    }
+    return 0.;
+  }
+
+  /**
+   * Computes the value of Soft Shrink activation function using a dense matrix
+   * as input.
+   * 
+   * @param x Input data.
+   * @param y The resulting output activation.
+   */
+  template<typename eT>
+  void Fn(const arma::Mat<eT>& x, arma::Mat<eT>& y)
+  {
+    y = x;
+    y.transform([](const double x){
+                  return Fn(x);});
+  }
+
+  /**
+   * Computes the inverse of the Soft Shrink function
+   *             | y + lambda, y > 0
+   * f^{-1}(y) = | y - lambda, y < 0
+   *             | 0, otherwise
+   * @param y
+   * @return f^{-1}(y)
+   */
+  double Inv(const double y)
+  {
+    if ( y > 0)
+    {
+      return y + lambda;
+    }
+    else if (y < 0)
+    {
+      return y - lambda;
+    }
+    return 0;
+  }
+
+  /**
+   * Computes the inverse of the Soft Shrink function
+   * 
+   * @param y Input data.
+   * @param x The resulting inverse of the input data
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Inv(const InputVecType& y,
+                  OutputVecType& x,
+                  const double lambda = 0.5)
+  {
+    x = y;
+    x.transform([](const double cx){
+                  return Inv(cx);});
+  }
+
+  /**
+   * Computes the first derivate of the Soft Shrink function
+   * 
+   * @param y Input activation
+   * @return f'(x) where f(x) = y
+   */
+  double Deriv(const double y)
+  {
+    if (y == 0)
+      return 0;
+    return 1;
+  }
+
+  /**
+   * Computes the first derivative of the Soft Shrink function.
+   *
+   * @param y Input activations.
+   * @param x The resulting derivatives. Should be the matrix used to
+   *          calculate activation y. 
+   * @param lambda parameter, default value = 0.5
+   */
+  template<typename InputVecType, typename OutputVecType>
+  void Deriv(const InputVecType& y,
+                    OutputVecType& x)
+  {
+    x = y;
+    x.transform([](const double cy){
```

Comment:
Looks like this fits on the same line.

Reply:
I have adopted new style for this when I switched to layer based function. Kindly have a look and suggest any changes.
Thanks.